### DISCUSSION TITLE
Roadmap comments

### DISCUSSION THREAD

**Discussion by mburns08109** [twinbasic/twinbasic#1062] [General]
Just a suggestion, but I'd consider flipping the priority of multi-platform (mostly because I'm afraid it will be a much bigger project than it appears) and the Edit & Continue implementation.

E&C will get us closer to the 1.0 shipping list than the multi-platform stuff will, being my thinking in this.

**Comment by WaynePhillipsEA** [comment]
I don't consider E&C as being an issue that prevents a v1.0 release.  Don't get me wrong, it's an important feature, but it's not a showstopper for v1.0.

**Reply by mburns08109** [reply]
Microsoft didn't think E&C was a "stop the presses" issue either in the VB.Net 1.0 days, but based upon the great caterwauling that arose due to its lack when vb.net 1.0 shipped without it, I'd suggest 2 things: 1) It has become one of those Visual Basic "essentials", and 2) that the lack of it in VB.Net 1.0 significantly slowed the acceptance of vb.net as "the direct replacement for VB6" that it "was supposed to be" (disregarding that whole COM/deterministic finalization stuff, of course).

Even today, VB.Net is widely regarded as "more the red-headed step-child" of the .Net world, and I think the lack of E&C in version 1.0 of vb.net contributed to that reputation to a significant degree. (says any "serious developer" of today: "Oh, you're a vb.net coder - how quaint!")

**Comment by VB6-Programming** [comment]
I'd also add that support for ActiveXs with unmatched bitness (32bit ActiveXs on 64 bit compiles) should have a higher priority.

(Not that I'm against the multi-platform stuff, I could do with the Android version TODAY ;) but I'm also concerned that multi-platform could take longer than expected).

**Comment by fafalone** [comment]
Is multi-platform even possible without making the language dependent on runtimes or frameworks?

**Reply by GaryMiller** [reply]
The V Programming Language has managed to make it work.

**Reply by mburns08109** [reply]
interesting - but how come I've never even heard of the V programming language until right now (resists the urge to make a p0rn pun here)?

**Reply by GaryMiller** [reply]
Not sure. I've been following it's development for a couple years now and Version 0.3 is just out.

It's quite usable though.  Compiles to tc very, very, fast.  And to other optimized C compilers for final optimized executable.

It has a built in GUI and ORM that works with several relational databases already.

They just passed 3000 stars on GitHub too.

https://vlang.io/

![image](https://user-images.githubusercontent.com/11558721/182190518-52c23c9c-1f00-4d84-8584-b53b772db03d.png)

**Reply by VBFNiya** [reply]
> Is multi-platform even possible without making the language dependent on runtimes or frameworks?

Nope. You need a runtime to ensure consistent low-level behavior across all platforms and you need a framework if you want "write once, run anywhere". If you try to create a multi-platform language without a runtime and framework you either end up with something extremely limited or you end up with C/C++ where half your code is compiler directives and needs to be compiled multiple times, one for each platform.

Wayne has proven himself a very smart guy. I won't be surprised if he has some kind of solution that compromises adequately between the leanness of no runtime/framework with the agility of having a runtime/framework.
